HWA-CQRS-API/Application/Rules/AuthBusinessRules.cs

```diff
@@ -0,0 +1,33 @@
+using Application.Constants;
+using Core.Application.Rules;
+using Core.Persistence.Repositories;
+using Domain.Models;
+using Persistence.Contexts;
+
+namespace Application.Rules;
+
+public class AuthBusinessRules(IRepository<User, AppDbContext> repository) : BaseBusinessRules
+{
+    private readonly IRepository<User, AppDbContext> _repository = repository;
+    
+    public Task UserShouldBeExists(User? user)
+    {
+        if (user == null)
+            throw new Exception(AuthMessages.UserDontExists);
+        return Task.CompletedTask;
+    }
+    
+    public async Task UserEmailShouldBeNotExists(string email)
+    {
+        User? user = await _repository.GetSingleOrDefaultAsync(predicate: u => u.Email == email, enableTracking: false);
+        if (user != null)
+            throw new Exception(AuthMessages.UserMailAlreadyExists);
+    }
+
+    public async Task UserPasswordShouldBeMatch(int id, string password)
+    {
+        User? user = await _repository.GetSingleOrDefaultAsync(predicate: u => u.Id == id, enableTracking: false);
```

Comment:
SingleOrDefault tüm userları tarayacağı için belki FirstOrDefault kullanılabilir. Çok büyük verilerde performans farkı yaratabilir.

Reply:
GetFirstOrDefaultAsyn metodu kullanıldı